### ISSUE TITLE
[Suggestion] Add variables to the gulpfile to store folder paths

### ISSUE THREAD

**Issue by NigelOToole** [yeoman/generator-webapp#472]
This would make changing folder names and/or locations easier. It would also avoid duplication of folder paths especially if the gulpfile is expanded further by the user with more tasks. This could take be something like the below.

```
const folders = {
  src:      'app',
  dest:     'dist',
  tmp:      '.tmp',
  fonts:    'fonts',
  images:   'images',
  scripts:  'scripts',
  styles:   'styles'
};
var paths = {
  src: {
    fonts:    folders.src + '/' + folders.fonts,
    images:   folders.src + '/' + folders.images,
    scripts:  folders.src + '/' + folders.scripts,
    styles:   folders.src + '/' + folders.styles
  },
  dest: {
    fonts:    folders.dest + '/' + folders.fonts,
    images:   folders.dest + '/' + folders.images,
    scripts:  folders.dest + '/' + folders.scripts,
    styles:   folders.dest + '/' + folders.styles
  }
};
```

This could then be used in tasks like.

`return gulp.src(paths.src.styles + '*.scss')`

**Comment by silvenon** [comment]
This has been suggested a few times, see the discussion in #295.

TL;DR in my opinion the gulpfile is too simple to introduce variables, it's not repetitive enough to be worth it. If you have a use case you can show me step by step where having variables would save you trouble, I'm listening :ear:

**Comment by NigelOToole** [comment]
Reading the discussion I understand your reasons for not doing this as the paths are not repeated that frequently. Where I think this would be most usefull is in either of the following use cases.

1. The user adds new tasks or extends current tasks, which would lead to the paths being used again.

**Comment by NigelOToole** [comment]
Reading the discussion I understand your reasons for not doing this as the paths are not repeated that frequently and could be replaced using a simple Find and Replace. 

However I do think having it configurable as a variable in the gulp file is a cleaner approach and it makes the gulp file more easily extendable. Where I think this would be most useful is in either of the following use cases.
1. The user adds new tasks or extends current tasks, which would lead to the paths being repeated more frequently.
2. The user wants to use this generator and integrate it with a particular CMS or back-end language which then dictates a different folder structure.
